zio-query/shared/src/main/scala/zio/query/ZQuery.scala

```diff
@@ -474,23 +495,44 @@ object ZQuery {
   def foreachPar[R, E, A, B, Collection[+Element] <: Iterable[Element]](
     as: Collection[A]
   )(f: A => ZQuery[R, E, B])(implicit bf: BuildFrom[Collection[A], B, Collection[B]]): ZQuery[R, E, Collection[B]] =
-    if (as.isEmpty) ZQuery.succeed(bf.newBuilder(as).result)
-    else {
-      val iterator                                         = as.iterator
-      var builder: ZQuery[R, E, Builder[B, Collection[B]]] = null
-      while (iterator.hasNext) {
-        val a = iterator.next()
-        if (builder eq null) builder = f(a).map(bf.newBuilder(as) += _)
-        else builder = builder.zipWithPar(f(a))(_ += _)
-      }
-      builder.map(_.result())
-    }
+    new ZQuery(cb => {
```

Comment:
This needs to be cleaned up to deal with some corner cases but with this change Caliban now beats Sangria on the complex query benchmark from the original issue.

```
[info] Benchmark                          Mode  Cnt    Score   Error  Units
[info] GraphQLBenchmarks.complexCaliban  thrpt    5  105.649 ± 2.733  ops/s
[info] GraphQLBenchmarks.complexSangria  thrpt    5  103.878 ± 3.786  ops/s
```